addons/mail/static/src/components/message_list/message_list.js

```diff
@@ -132,6 +132,9 @@ class MessageList extends Component {
                 case 'more-messages-loaded':
                     this._adjustFromMoreMessagesLoaded(hint);
                     break;
+                case 'scroll-position-saved':
+                    this._adjustFromScollPositionSaved(hint);
+                    break;
```

Comment:
This is assuming that we are only saving the scroll position when we know we will want to restore it.
At least that's what the name of the hint implies

With the current name, it feels like the hint should be added in `saveThreadCacheScrollPositionsAsInitial`.

But I think it is an incorrect assumption in general. In an ideal world (with no perf issue) we would want to save the new scroll position whenever it changes, however we would only want to restore it in specific cases, and not to consume the hint when the next random re-render happens.

So I think we do want to keep it flux specific as it is, but it should probably be renamed, and it's currently not in the appropriate flux I think.

The question to ask is "when do we want to restore the scroll", and the answer is not "when a window is folded" or "when a window is hidden by main menu" but the opposite.

We want to restore scroll when the chat window is unfolded, or shown from whatever reason. If we were on an ideal world (the one where it is saved at every change), we would want to restore it at every render as well, without any hint required (render which would most of the time do nothing due to already having the correct value).
Since we are not on an ideal world and the correct value is not always in the store, we need to find the right match.
When I see the name of the other hint/handlers, it feels like the correct solution is to do it on:
- "adjust from chat window unfold"
- "adjust from home menu show"
- "adjust from initial render" (hint being set should be a default value of thread viewer somehow so that "initial" scroll position is actually taken care of "initially")
- others if necessary?

Reply:
Agree with @seb-odoo:

When scroll adjustments are well-designed, hint contains information on what just happens in models and components adjust scrolling (if any) accordingly. This is not the case here.

Also not correct to adjust scroll in direct response to a save: the adjustment of scrolls should happen from another event, be like "chat window (un)folded" or "home menu nuisance just happened".

Reply:
Didn't understand that part : 

> "adjust from initial render" (hint being set should be a default value of thread viewer somehow so that "initial" scroll position is actually taken care of "initially")

Think I made the rest

Reply:
`threadCacheInitialScrollPosition` value will only apply if there is a hint set to request it.

However, we currently don't set any hint by default. Which means the real initial value will never be taken into account.
I would expect that the very first render of any thread viewer would render with the `threadCacheInitialScrollPosition` correctly set automatically. To solve this issue, I think there should be a new hint, that is added by default to the list when creating a viewer.

Reply:
PS: maybe this is actually `change-of-thread-cache` which is also considering the "init" as "change", but then its name is slightly off.

Reply:
In a way, this is a change of thread cache as you went from "no cache" to "a cache", so the name is not that "off" but yeah it's not totally obvious that this hint is created at first render... Would you add an hint which would be triggered if previous.threadCache was undefined, named like "first-rendering" or I don't know .. ?